py-polars/polars/series/series.py

```diff
@@ -1635,257 +1637,421 @@ def to_dummies(self, separator: str = "_") -> DataFrame:
+        as_series: Literal[False],
+    ) -> DataFrame:
+        ...
 
+    @overload
+    def qcut(
+        self,
+        quantiles: Sequence[float] | int,
+        *,
+        labels: Sequence[str] | None = ...,
+        left_closed: bool = ...,
+        allow_duplicates: bool = ...,
+        include_breaks: bool = ...,
+        break_point_label: str = ...,
+        category_label: str = ...,
+        as_series: bool,
+    ) -> Series | DataFrame:
+        ...
+
+    @deprecate_renamed_parameter("series", "as_series", version="0.18.14")
     @deprecate_renamed_parameter("q", "quantiles", version="0.18.12")
     def qcut(
         self,
-        quantiles: list[float] | int,
+        quantiles: Sequence[float] | int,
         *,
-        labels: list[str] | None = None,
-        break_point_label: str = "break_point",
-        category_label: str = "category",
-        series: bool = True,
+        labels: Sequence[str] | None = None,
         left_closed: bool = False,
         allow_duplicates: bool = False,
         include_breaks: bool = False,
-    ) -> DataFrame | Series:
+        break_point_label: str = "break_point",
+        category_label: str = "category",
+        as_series: bool = True,
+    ) -> Series | DataFrame:
         """
-        Discretize continuous values into discrete categories based on their quantiles.
+        Bin continuous values into discrete categories based on their quantiles.
 
         Parameters
         ----------
         quantiles
             Either a list of quantile probabilities between 0 and 1 or a positive
-            integer determining the number of evenly spaced probabilities to use.
+            integer determining the number of bins with uniform probability.
         labels
-            Labels to assign to the quantiles. If given the length of labels must be
-            len(breaks) + 1.
-        break_point_label
-            Name given to the breakpoint column/field. Only used if series == False or
-            include_breaks == True
-        category_label
-            Name given to the category column. Only used if series == False.
-        series
-            If True, return a categorical Series in the data's original order
+            Names of the categories. The number of labels must be equal to the number
+            of cut points plus one.
         left_closed
-            Whether intervals should be [) instead of (]
+            Set the intervals to be left-closed instead of right-closed.
         allow_duplicates
-            If True, the resulting quantile breaks don't have to be unique. This can
-            happen even with unique probs depending on the data. Duplicates will be
-            dropped, resulting in fewer bins.
+            If set to ``True``, duplicates in the resulting quantiles are dropped,
+            rather than raising a `DuplicateError`. This can happen even with unique
+            probabilities, depending on the data.
```

Comment:
I think it's helpful to retain mention of the fact that this will result in fewer bins that expected to allay any surprise.